scala/spark-pubsub/README.md

```diff
@@ -0,0 +1,161 @@
+
+1. [Create a Cloud Bigtable cluster][create-cluster]. Be sure to note the
+project ID.
+2. [Create a service account and a JSON key file][json-key].
+
+[create-cluster]: https://cloud.google.com/bigtable/docs/creating-cluster
+[json-key]: https://cloud.google.com/bigtable/docs/installing-hbase-client#service-account
+
+
+## Creating Compute Engine VM Instances for Cloud Bigtable, Cloud Pubsub, and Spark using bdutil
+
+Please add the following two lines to the GCE_SERVICE_ACCOUNT_SCOPES in extentions/bigtable/bigtable_env.sh. The added scopes enable calling Cloud Pubsub API on the GCE VMs you are going to create.
+
+    'https://www.googleapis.com/auth/cloud-platform'
+    'https://www.googleapis.com/auth/pubsub'
+
+Create new GCE VM instance with Hadoop, Spark, and Cloud Bigtable
+
+    $ git clone https://github.com/GoogleCloudPlatform/bdutil.git
+    $ cd bdutil
+    $ ./bdutil -e hadoop2_env.sh -e extensions/spark/spark_env.sh -e extensions/bigtable/bigtable_env.sh -e [path/to/config/file.sh] -f deploy
+
+
+
+## Overview of the code sample
+
+There are three components in this example: the Spark-Cloud Pubsub connector, the message producer, and the message processor.
+
+The Spark-Cloud Pubsub connector (in the spark-cloud-pubsub-connector directory) contains three files: CloudPubsubInputDStream.scala, CloudPubsubUtils.scala, and RetryHttpInitializerWrapper.scala. CloudPubsubInputDStream.scala contains the CloudPubsubInputDStream class that extends the InputDStream class in Spark. This enables pulling messages from a Cloud Pubsub topic in the time interval that a user specifies when he/she instantiates a Spark streaming context object in their Spark application:
+
+    //from https://github.com/taragu/cloud-bigtable-examples/blob/pubsub/scala/spark-pubsub-wordcount/cloud-pubsub-receiver/src/main/scala/CloudPubsubReceiver.scala#L72
+    val ssc = new StreamingContext(sparkConf, Seconds(samplingFreq.toInt))
+    
+
+You can instantiate a CloudPubsubInputDStream object with your Cloud Pubsub information:
+    
+    //from https://github.com/taragu/cloud-bigtable-examples/blob/pubsub/scala/spark-pubsub-wordcount/cloud-pubsub-receiver/src/main/scala/CloudPubsubReceiver.scala#L97
+    val ackIDMessagesDStream = CloudPubsubUtils.createDirectStream(ssc, projectName, topicName, subscriptionName)
+
+CloudPubsubUtils.scala contains utility methods that can be used either in the Spark-Cloud Pubsub connector (listSubscriptions, and getClient) or the message processor (sendAcks, and createDirectStream). 
+
+RetryHttpInitializerWrapper.scala retries failed RPC calls, and is called in the getClient method in CloudPubsubUtils.scala
+
+You can use the Spark-Cloud Pubsub connector in two ways. If you would like to use it without any modification, you can download a pre-built jar from a GCS storage bucket with the following command:
+
+    $ gsutil cp gs://cloud-bigtable-examples/spark-cloud-pubsub-connector_2.10-0.0.jar PATH/TO/SAVE/THE/FILE
```

Comment:
@sduskis Is this really where / how we wish to distribute this?  Shouldn't we add it to our mvn repo?